verilog/CST/verilog_tree_print.cc

```diff
@@ -70,4 +72,40 @@ void PrettyPrintVerilogTree(const verible::Symbol& root, absl::string_view base,
   root.Accept(&printer);
 }
 
+
+VerilogTreeToJsonConverter::VerilogTreeToJsonConverter(absl::string_view base)
+    : context_(verible::TokenInfo::Context(base)),
+    json_(Json::objectValue), value_(&json_) {}
+
+void VerilogTreeToJsonConverter::Visit(const verible::SyntaxTreeLeaf& leaf) {
+  *value_ = Json::objectValue;
+  (*value_)["symbol"] = verilog_symbol_name(leaf.get().token_enum());
+  (*value_)["start"]  = leaf.get().left(context_.base);
+  (*value_)["end"]    = leaf.get().right(context_.base);
+}
+
+void VerilogTreeToJsonConverter::Visit(const verible::SyntaxTreeNode& node) {
+  *value_ = Json::objectValue;
+  (*value_)["tag"] = NodeEnumToString(static_cast<NodeEnum>(node.Tag().tag));
+  Json::Value &children = (*value_)["children"] = Json::arrayValue;
+  children.resize(node.children().size());
+
+  {
```

Comment:
Comment here that you're intentionally preserving null slots from the original CST.

Reply:
Done.